lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -0,0 +1,778 @@
+    required: false
+  security_groups:
+    description:
+      - A list of the names or IDs of the security groups to assign to the load balancer. Required if state=present.
+    required: false
+    default: []
+  scheme:
+    description:
+      - Internet-facing or internal load balancer. An ELB scheme can not be modified after creation.
+    required: false
+    default: internet-facing
+    choices: [ 'internet-facing', 'internal' ]
+  state:
+    description:
+      - Create or destroy the load balancer.
+    required: true
+    choices: [ 'present', 'absent' ]
+  tags:
+    description:
+      - A dictionary of one or more tags to assign to the load balancer.
+    required: false
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Create an ELB and attach a listener
+- elb_application_lb:
+    name: myelb
+    security_groups:
+      - sg-12345678
+      - my-sec-group
+    subnets:
+      - subnet-012345678
+      - subnet-abcdef000
+    listeners:
+      - protocol: http # Required. The protocol for connections from clients to the load balancer (http or https).
+        port: 80 # Required. The port on which the load balancer is listening.
+        ssl_policy: # The security policy that defines which ciphers and protocols are supported. The default is the current predefined security policy.
+        certificates: # The ARN of the certificate (only one certficate ARN should be provided)
+        default_actions:
+          - type: forward # Required. Only 'forward' is accepted at this time
+            target_group_arn: # Required. The ARN of the target group
+    state: present
+
+# Create an ELB and attach a listener and attributes
+- elb_application_lb:
+    name: myelb
+    security_groups:
+      - sg-12345678
+      - my-sec-group
+    subnets:
+      - subnet-012345678
+      - subnet-abcdef000
+    listeners:
+      - protocol: http # Required. The protocol for connections from clients to the load balancer (http or https).
+        port: 80 # Required. The port on which the load balancer is listening.
+        ssl_policy: # The security policy that defines which ciphers and protocols are supported. The default is the current predefined security policy.
+        certificates: # The ARN of the certificate (only one certficate ARN should be provided)
+        default_actions:
+          - type: forward # Required. Only 'forward' is accepted at this time
+            target_group_arn: # Required. The ARN of the target group
+    attributes:
+        - Key: idle_timeout.timeout_seconds
+          Value: '60'
+    state: present
+
+# Create an ALB with listeners and rules
+- elb_lb_application:
+    name: test-alb
+    subnets:
+      - subnet-12345678
+      - subnet-87654321
+    security_groups:
+     - sg-12345678
+    scheme: internal
+    listeners:
```

Comment:
This is because it's passing everything directly to boto3 for listeners. I've seen similar in other modules